### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by FilipDolnik**

### CODE REVIEW

File: AlamofireRequestPerformer/AlamofireRequestPerformer.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+//  Created by Tadeas Kriz on 28/07/15.
+//  Copyright © 2015 Brightify. All rights reserved.
+//
+
+import Alamofire
+import HTTPStatusCodes
+
+public struct AlamofireRequestPerformer: RequestPerformer {
+    
+    public let dataEncoder: DataEncoder
```
Comment (TadeasKriz) [inline AlamofireRequestPerformer/AlamofireRequestPerformer.swift:14]: Should it be public?

File: ObjectMapper/Map/DeserializableData.swift [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+//  Created by Filip Dolnik on 21.10.16.
+//  Copyright © 2016 Brightify. All rights reserved.
+//
+
+public struct DeserializableData {
+    
+    public let data: SupportedType
+    public let polymorph: Polymorph?
+    
+    public init(data: SupportedType, polymorph: Polymorph?) {
```
Comment (TadeasKriz) [inline ObjectMapper/Map/DeserializableData.swift:14]: Should it know about the `Polymorph`?

File: ObjectMapper/Map/DeserializableData.swift [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    
+    public func get<T, R: DeserializableTransformation>(using transformation: R, or: [String: T?]) -> [String: T?] where R.Object == T {
+        return get(using: transformation) ?? or
+    }
+    
+    public func get<T, R: DeserializableTransformation>(using transformation: R) throws -> [String: T?] where R.Object == T {
+        return try valueOrThrow(get(using: transformation))
+    }
+    
+    public func deserialize<T: Deserializable>(_ type: SupportedType) -> T? {
```
Comment (TadeasKriz) [inline ObjectMapper/Map/DeserializableData.swift:197]: This implementation should be in ObjectMapper, not in the DeserializableData

File: ObjectMapper/Map/DeserializableMappableDataWrapper.swift [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    public init(delegate: DeserializableData) {
+        self.delegate = delegate
+    }
+    
+    public subscript(path: [String]) -> MappableData {
+        get {
+            return DeserializableMappableDataWrapper(delegate: delegate[path])
+        }
+        set {
+        }
```
Comment (TadeasKriz) [inline ObjectMapper/Map/DeserializableMappableDataWrapper.swift:22]: If this is intentionally left empty, there needs to be a comment about it (for example: `// Intentionally left empty to conform protocol requirements.`)

File: AlamofireRequestPerformer/AlamofireRequestPerformer.swift (hunk 1 above)
Comment (FilipDolnik) [reply AlamofireRequestPerformer/AlamofireRequestPerformer.swift:14]: It`s from RequestPerformer protocol.

**Pull request merged**